sys-utils/hwclock.c

```diff
@@ -246,12 +246,21 @@ static int read_adjtime(const struct hwclock_control *ctl,
 
 	fclose(adjfile);
 
-	sscanf(line1, "%lf %ld %lf",
-	       &adjtime_p->drift_factor,
-	       &adjtime_p->last_adj_time,
-	       &adjtime_p->not_adjusted);
+	if (sizeof(time_t) > 4) {
+		sscanf(line1, "%lf %lld %lf",
+			&adjtime_p->drift_factor,
+			(int64_t*)&adjtime_p->last_adj_time,
+			&adjtime_p->not_adjusted);
 
-	sscanf(line2, "%ld", &adjtime_p->last_calib_time);
+		sscanf(line2, "%lld", (int64_t*)&adjtime_p->last_calib_time);
+	} else {
+		sscanf(line1, "%lf %ld %lf",
+			&adjtime_p->drift_factor,
+			(long*)&adjtime_p->last_adj_time,
+			&adjtime_p->not_adjusted);
+
+		sscanf(line2, "%ld", (long*)&adjtime_p->last_calib_time);
+	}
```

Comment:
Would be better to use locally only 64-bit numbers for scanf() and then cast to time_t ?
```C
int64_t last;

sscanf(line1, "%lf %"SCNs64"%lf", adjtime_p->drift_factor, &last, &adjtime_p->not_adjusted);
adjtime_p->last_adj_time = (time_t) last;
```
it seems less code. Note also SCNs64.